src/properties/struct_property.rs

```diff
@@ -91,22 +98,44 @@ impl StructProperty {
 
         let value = match type_name.as_str() {
-            "Vector" => StructPropertyValue::Vector(Vector::new(
-                FloatProperty::read(cursor, false)?.value.0,
-                FloatProperty::read(cursor, false)?.value.0,
-                FloatProperty::read(cursor, false)?.value.0,
-            )),
-            "Rotator" => StructPropertyValue::Rotator(Rotator::new(
-                FloatProperty::read(cursor, false)?.value.0,
-                FloatProperty::read(cursor, false)?.value.0,
-                FloatProperty::read(cursor, false)?.value.0,
-            )),
-            "Quat" => StructPropertyValue::Quat(Quat::new(
-                FloatProperty::read(cursor, false)?.value.0,
-                FloatProperty::read(cursor, false)?.value.0,
-                FloatProperty::read(cursor, false)?.value.0,
-                FloatProperty::read(cursor, false)?.value.0,
-            )),
+            "Vector" => match large_world_coordinates {
+                true => StructPropertyValue::VectorD(VectorD::new(
+                    DoubleProperty::read(cursor, false)?.value.0,
+                    DoubleProperty::read(cursor, false)?.value.0,
+                    DoubleProperty::read(cursor, false)?.value.0,
+                )),
+                false => StructPropertyValue::VectorF(VectorF::new(
+                    FloatProperty::read(cursor, false)?.value.0,
+                    FloatProperty::read(cursor, false)?.value.0,
+                    FloatProperty::read(cursor, false)?.value.0,
+                )),
+            },
+            "Rotator" => match large_world_coordinates {
+                true => StructPropertyValue::RotatorD(RotatorD::new(
+                    DoubleProperty::read(cursor, false)?.value.0,
+                    DoubleProperty::read(cursor, false)?.value.0,
+                    DoubleProperty::read(cursor, false)?.value.0,
+                )),
+                false => StructPropertyValue::RotatorF(RotatorF::new(
+                    FloatProperty::read(cursor, false)?.value.0,
+                    FloatProperty::read(cursor, false)?.value.0,
+                    FloatProperty::read(cursor, false)?.value.0,
+                )),
+            },
+            "Quat" => match large_world_coordinates {
+                true => StructPropertyValue::QuatD(QuatD::new(
+                    DoubleProperty::read(cursor, false)?.value.0,
+                    DoubleProperty::read(cursor, false)?.value.0,
+                    DoubleProperty::read(cursor, false)?.value.0,
+                    DoubleProperty::read(cursor, false)?.value.0,
+                )),
+                false => StructPropertyValue::QuatF(QuatF::new(
+                    FloatProperty::read(cursor, false)?.value.0,
+                    FloatProperty::read(cursor, false)?.value.0,
+                    FloatProperty::read(cursor, false)?.value.0,
+                    FloatProperty::read(cursor, false)?.value.0,
+                )),
+            },
```

Comment:
This version of the code is more illustrative of the tradeoffs of using `LittleEndian::read_fXX` and `::write_fXX` vs `*Property::read` and `::write`